### PR TITLE
Create test-specific 🐳 file

### PR DESCRIPTION
**Pull request by michplunkett**
## Related to:
- https://github.com/lucyparsons/OpenOversight/issues/1054

## Description of Changes
Creating a test-specific 🐳 Docker file. This will be the first step in implementing PostgreSQL in the testing environment so that we can get a true 1:1 comparison between the environments.

## Tests and linting
 - [x] This branch is up-to-date with the `develop` branch.
 - [x] `pytest` passes on my local development environment.
 - [x] `pre-commit` passes on my local development environment.

### CODE REVIEW

File: dockerfiles/web/Dockerfile [hunk 1]
```diff
@@ -39,19 +28,14 @@ COPY yarn.lock /usr/src/app/
...
 RUN pip3 install --no-cache-dir -r requirements.txt
 
-RUN test "${DOCKER_BUILD_ENV}" = production || pip3 install --no-cache-dir -r dev-requirements.txt
-
 COPY package.json /usr/src/app/
 RUN yarn
 
 COPY test_data.py /usr/src/app/
-COPY mypy.ini /usr/src/app/
-EXPOSE 3000
```
Comment (michplunkett) [inline dockerfiles/web/Dockerfile:52]: This line was redundant.

File: .github/workflows/test_prs.yml [hunk 2]
```diff
@@ -1,28 +1,19 @@
 name: CI
 
-# Controls when the action will run.
 on:
   pull_request:
     branches:
       - develop
-      - main
```
Comment (michplunkett) [inline .github/workflows/test_prs.yml:8]: We have stopped using `main`, so I removed it from this file.

File: dockerfiles/web/Dockerfile-test [hunk 3]
```diff
@@ -40,21 +39,14 @@ RUN chmod -R 777 /usr/src/app/ /var/lib/xkb /.cache /.yarn /.mozilla
...
-
-RUN test "${DOCKER_BUILD_ENV}" = production || pip3 install --no-cache-dir -r dev-requirements.txt
+RUN pip3 install --no-cache-dir -r requirements.txt && pip3 install --no-cache-dir -r dev-requirements.txt
 
 COPY package.json /usr/src/app/
 RUN yarn
 
 COPY test_data.py /usr/src/app/
-COPY mypy.ini /usr/src/app/
-EXPOSE 3000
```
Comment (michplunkett) [inline dockerfiles/web/Dockerfile-test:52]: This is redundant with the docker-compose file.

**Review by b-meson — APPROVED** [review]

**Comment by b-meson** [comment]
Thanks for working on this @michplunkett, feel free to merge it whenever you can.

**Comment by michplunkett** [comment]
I'm workin' on some school stuff now, but I'll likely merge it tomorrow when I can make sure everything is 💯 .

**Comment by michplunkett** [comment]
I ran everything on my local again, and it looks like we're good to go.

**Pull request merged**